middleware/authentication_test.go

```diff
@@ -0,0 +1,106 @@
+package middleware
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/lvl484/user-manager/logger"
+	"github.com/lvl484/user-manager/model"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	logger.SetLogger(&logger.LogConfig{Output: "Stdout", Level: "debug"})
+
+	code := m.Run()
+
+	os.Exit(code)
+}
+
+func TestBasicAuthenticationMiddlewareValidPass(t *testing.T) {
+	ma := &MockAuth{}
+	ba := NewBasicAuthentication(ma)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	r.SetBasicAuth("i3odja", "1q2w3e4r")
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusOK, w.Code)
+}
+
+func TestBasicAuthenticationMiddlewareInvalidPass(t *testing.T) {
+	ma := &MockAuth{}
+	ba := NewBasicAuthentication(ma)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	r.SetBasicAuth("i3odja", "123123")
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	checkErrorResponse(t, w, http.StatusUnauthorized)
+}
+
+func TestBasicAuthenticationMiddlewareError(t *testing.T) {
+	ma := &MockAuth{Error: errors.New("middleware error")}
+	ba := NewBasicAuthentication(ma)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	r.SetBasicAuth("i3odja", "123123")
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	checkErrorResponse(t, w, http.StatusInternalServerError)
+}
+
+func checkErrorResponse(t *testing.T, w *httptest.ResponseRecorder, expectedCode int) {
+	mError := model.Error{}
+
+	err := json.Unmarshal(w.Body.Bytes(), &mError)
+	require.NoError(t, err)
+
+	assert.NotEmpty(t, mError.Code)
+	assert.NotEmpty(t, mError.Message)
+
+	assert.Equal(t, expectedCode, w.Code)
+}
+
+type MockAuth struct {
```

Comment:
Let's start using mock generation https://github.com/golang/mock

Reply:
Thank you for advice. I fixed it